.travis.yml

```diff
@@ -16,12 +25,30 @@ before_install:
   - sleep 3
   - export DISPLAY=:1.0
   - ./TeensyduinoInstall.linux64 --dir=$ARDUINO_PATH
+  
+  # Adapted from https://github.com/felixduvallet/ros-travis-integration/blob/master/.travis.yml
+  - sudo sh -c "echo \"deb http://packages.ros.org/ros/ubuntu xenial main\" > /etc/apt/sources.list.d/ros-latest.list"
+  - sudo apt-key adv --keyserver 'hkp://keyserver.ubuntu.com:80' --recv-key C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654
+  - sudo apt-get update -qq
+  - sudo apt-get install dpkg
+  - sudo apt-get install -y python-catkin-pkg python-rosdep python-wstool ros-kinetic-ros-base
+  - source /opt/ros/kinetic/setup.bash
+  # Prepare rosdep to install dependencies.
+  - sudo rosdep init
+  - rosdep update 
 install:
-  - pip install -r requirements.txt
+  - pip install -r requirements.txt -r requirements-dev.txt
   - pip install -e .
   - python setup.py develop
+  
+  # Set up and initialize catkin workspace
+  - cd $CI_SOURCE_PATH/robot/rospackages/src
+  - cd .. && catkin_make
+  - source devel/setup.bash
+before_script:
+  - rosdep install -y -r --from-paths src --ignore-src
 script:
-  - cd robot/rover/build
+  - cd $CI_SOURCE_PATH/robot/rover/build
   - cmake ..
   - make
   - cd ../../..
```

Comment:
Assuming `CI_SOURCE_PATH` is repo root, we can change this to `cd $CI_SOURCE_PATH`, or `cd $ROOT_REPO` if you agree to rename the variable

Reply:
changed